include/boost/json/impl/monotonic_resource.ipp

```diff
@@ -97,17 +151,25 @@ do_allocate(
     std::size_t n,
     std::size_t align)
 {
+    // use the initial block if possible
+    if(initial_.base)
+        if(auto p = allocate_in_block(initial_, n, align))
+            return p;
+    // otherwise, try allocate in the current block
     if(head_)
-    {
-        auto p = head_->alloc(n, align);
-        if(p)
+        if(auto p = allocate_in_block(*head_, n, align))
             return p;
-    }
-    if(n > block_size_ - 2 * align)
-        alloc_block(n + 2 * align);
-    else
-        alloc_block(block_size_);
-    auto p = head_->alloc(n, align);
+    // neither of the above was sucessful,
+    // so we need to allocate a new block.
```

Comment:
This comment adds no information

Reply:
Those comments just walk through the process -- makes it much faster to understand what's going on